paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/FormData.kt

```diff
@@ -0,0 +1,31 @@
+package com.stripe.android.paymentsheet.forms
+
+class FormData(
+    val source: Map<String, Any?>,
+    val elementKeys: Map<String, String?>
+) {
+    fun toMap(): MutableMap<String, Any?> {
+        val destMap = mutableMapOf<String, Any?>()
+        createMap(source, destMap, elementKeys)
+        return destMap
+    }
+
+    companion object {
+        @Suppress("UNCHECKED_CAST")
+        private fun createMap(
```

Comment:
Can you add a comment explaining at a high level the logic here?

Reply:
done